File: ios/FreelanceKar/Store/AppState.swift
import Foundation

/// Mirrors src/lib/types.ts's AppState shape field-for-field.
struct AppState {
    // Navigation
    var screen: Screen = .home
    var freelancerMode: Bool = false

    // Discovery
    var cityIdx: Int = 0
    var locality: String = ""
    var cityOpen: Bool = false
    var citySearch: String = ""
    var query: String = ""
    var brief: String = ""
    var searchLabel: String = ""
    var searching: Bool = false
    var sort: Sort = .match
    var filters = FilterSet()
    var applied = FilterSet()
    var filtersOpen: Bool = false

    // Freelancer detail & hiring
    var activeId: String? = nil
    var svcIdx: Int = 0
    var saved: [String] = []
    var hire: Int = 0 // 0 closed, 1 details, 2 pay, 3 success, 4 failure
    var hireNote: String = ""
    var hireDateIdx: Int = 0
    var hireError: Bool = false
    var methodIdx: Int = 0
    var paying: Bool = false
    var payRetried: Bool = false

    // Orders & messaging
    var bookings: [Booking] = []
    var viewOrder: String? = nil
    var threads: [ChatThread] = []
    var threadId: String? = nil
    var draft: String = ""
    var typing: Bool = false

    // Disputes & reviews
    var issueOpen: Bool = false
    var issueReason: Int = 0
    var issueText: String = ""
    var issueError: Bool = false
    var reviewOpen: Bool = false
    var reviewStars: Int = 0
    var reviewText: String = ""
    var reviewError: Bool = false

    // Notifications
    var notifs: [Notif] = Seed.seedNotifs

    // Freelancer onboarding & account
    var onbStep: Int = 1
    var phone: String = ""
    var otpSent: Bool = false
    var otp: String = ""
    var onbErrorMsg: String = ""
    var fName: String = ""
    var fCityIdx: Int = 0
    var fRadiusIdx: Int = 1
    var fBio: String = ""
    var kycAadhaar: Bool = false
    var kycPan: Bool = false
    var kycSelfie: Bool = false
    var bank: Bool = false
    var svcRows: [ServiceRow] = [ServiceRow(cat: 0, type: 0, amount: "")]
    var onboarded: Bool = false
    var subscribed: Bool = false
    var planIdx: Int = 1
    var kycRejected: Bool = false

    // Freelancer working state
    var available: Bool = true
    var reqs: [JobRequest] = Seed.reqs
    var quotesSent: Int = 0
    var quoteFor: String? = nil
    var quoteAmount: String = ""
    var quoteDayIdx: Int = 1
    var quoteMsg: String = ""
    var quoteError: Bool = false
    var fGigs: [Gig] = Seed.seedGigs
    var payments: [Payment] = Seed.payments

    // UI
    var toast: String = ""

    var kycCount: Int { [kycAadhaar, kycPan, kycSelfie].filter { $0 }.count }
}

/// Perceived rhythm of the prototype (src/lib/store.tsx TIMING) — keep these
/// values when wiring real network calls.
enum TIMING {
    static let search: UInt64 = 750_000_000
    static let payConfirm: UInt64 = 1_100_000_000
    static let accept: UInt64 = 3_200_000_000
    static let deliver: UInt64 = 9_000_000_000
    static let chatReply: UInt64 = 1_600_000_000
    static let supportReply: UInt64 = 4_500_000_000
    static let toast: UInt64 = 2_200_000_000
    static let profileLive: UInt64 = 1_200_000_000
    static let subscribed: UInt64 = 600_000_000
}

File: android/app/src/main/java/com/freelancekar/app/data/AppState.kt
package com.freelancekar.app.data

/** Mirrors src/lib/types.ts AppState — the single app-wide state shape. */
data class AppState(
    // Navigation
    val screen: Screen = Screen.HOME,
    val freelancerMode: Boolean = false,

    // Discovery
    val cityIdx: Int = 0,
    val locality: String = "",
    val cityOpen: Boolean = false,
    val citySearch: String = "",
    val query: String = "",
    val brief: String = "",
    val searchLabel: String = "",
    val searching: Boolean = false,
    val sort: Sort = Sort.MATCH,
    val filters: FilterSet = FilterSet(),
    val applied: FilterSet = FilterSet(),
    val filtersOpen: Boolean = false,

    // Freelancer detail & hiring
    val activeId: String? = null,
    val svcIdx: Int = 0,
    val saved: List<String> = emptyList(),
    val hire: Int = 0, // 0 closed, 1 details, 2 pay, 3 success, 4 failure
    val hireNote: String = "",
    val hireDateIdx: Int = 0,
    val hireError: Boolean = false,
    val methodIdx: Int = 0,
    val paying: Boolean = false,
    val payRetried: Boolean = false,

    // Orders & messaging
    val bookings: List<Booking> = emptyList(),
    val viewOrder: String? = null,
    val threads: List<Thread> = emptyList(),
    val threadId: String? = null,
    val draft: String = "",
    val typing: Boolean = false,

    // Disputes & reviews
    val issueOpen: Boolean = false,
    val issueReason: Int = 0,
    val issueText: String = "",
    val issueError: Boolean = false,
    val reviewOpen: Boolean = false,
    val reviewStars: Int = 0,
    val reviewText: String = "",
    val reviewError: Boolean = false,

    // Notifications
    val notifs: List<Notif> = SEED_NOTIFS,

    // Freelancer onboarding & account
    val onbStep: Int = 1,
    val phone: String = "",
    val otpSent: Boolean = false,
    val otp: String = "",
    val onbErrorMsg: String = "",
    val fName: String = "",
    val fCityIdx: Int = 0,
    val fRadiusIdx: Int = 1,
    val fBio: String = "",
    val kyc: Kyc = Kyc(),
    val bank: Boolean = false,
    val svcRows: List<ServiceRow> = listOf(ServiceRow(0, 0, "")),
    val onboarded: Boolean = false,
    val subscribed: Boolean = false,
    val planIdx: Int = 1,
    val kycRejected: Boolean = false,

    // Freelancer working state
    val available: Boolean = true,
    val reqs: List<JobRequest> = REQS,
    val quotesSent: Int = 0,
    val quoteFor: String? = null,
    val quoteAmount: String = "",
    val quoteDayIdx: Int = 1,
    val quoteMsg: String = "",
    val quoteError: Boolean = false,
    val fGigs: List<Gig> = SEED_GIGS,
    val payments: List<Payment> = PAYMENTS,

    // UI
    val toast: String = "",
    val toastToken: Long = 0L,
)

/** Timing constants mirroring src/lib/store.tsx's TIMING object (milliseconds). */
object Timing {
    const val SEARCH = 750L
    const val PAY_CONFIRM = 1100L
    const val ACCEPT = 3200L
    const val DELIVER = 9000L
    const val CHAT_REPLY = 1600L
    const val SUPPORT_REPLY = 4500L
    const val TOAST = 2200L
    const val PROFILE_LIVE = 1200L
    const val SUBSCRIBED = 600L
}
